godot-macros/src/godot_api.rs

```diff
@@ -301,48 +305,61 @@ fn extract_attributes<T>(
 where
     for<'a> &'a T: Spanned,
 {
-    let mut found = None;
-    for (index, attr) in attributes.iter().enumerate() {
-        let attr_name = attr
-            .get_single_path_segment()
-            .expect("get_single_path_segment");
-
-        let new_found = match attr_name {
-            name if name == "func" => Some(BoundAttr {
-                attr_name: attr_name.clone(),
-                index,
-                ty: BoundAttrType::Func(attr.value.clone()),
-            }),
-            name if name == "signal" => {
-                // TODO once parameters are supported, this should probably be moved to the struct definition
-                // E.g. a zero-sized type Signal<(i32, String)> with a provided emit(i32, String) method
-                // This could even be made public (callable on the struct obj itself)
-                Some(BoundAttr {
-                    attr_name: attr_name.clone(),
-                    index,
-                    ty: BoundAttrType::Signal(attr.value.clone()),
-                })
-            }
-            name if name == "constant" => Some(BoundAttr {
-                attr_name: attr_name.clone(),
-                index,
-                ty: BoundAttrType::Const(attr.value.clone()),
-            }),
-            _ => None,
-        };
+    const FUNC: &str = "func";
+    const SIGNAL: &str = "signal";
+    const CONSTANT: &str = "constant";
+
+    let mut name_and_parser = None;
 
-        // Validate at most 1 attribute
-        if found.is_some() && new_found.is_some() {
-            bail!(
-                &error_scope,
-                "at most one #[func], #[signal], or #[constant] attribute per declaration allowed",
-            )?;
+    let mut found_attr_count = 0;
```

Comment:
Could also check if `name_and_parser.is_some()` and bail out like that. Let me know which is preferable.

Reply:
Yes, I think that's better because the logic is more local. In the current code, I first have to see what happens with `found_attr_count`.